library(shiny)
source("globals.R")

fluidPage(
    shinyjs::useShinyjs(),
    shinyjs::inlineCSS(appCSS),
    titlePanel("Evaluation of Cancer Driver Prediction Methods"),
    div(id = "form",
        h3("General"),
        textInput("method_name", labelMandatory("Name of method")),
        numericInput("year_published", labelMandatory("Year of publication"), 2016, 1900, 2016),
        textInput("title", labelMandatory("Title of publication")),
        textInput("journal", labelMandatory("Journal")),
        textInput("authors", labelMandatory("Authors")),
        textInput("pubmedID", labelMandatory("PubMed ID")),
        radioButtons("scope", "Does the method/publication fit our scope",
                     c("Yes", "Maybe", "No")),
        hr(),
        h3("Input Data"),
        radioButtons("seq_data", "What type of genome sequencing data are used",
                     c("WES", "WGS", "Gene panels", "Other (please describe)" = "Other")),
        conditionalPanel("input.seq_data == 'Other'", textInput("seq_data", "Describe")),
        radioButtons("IDstrategy", "What is the primary strategy of identifying driver genes",
                     c("Recurrence (higher than expected number of mutations in region" ,
                       "Functional impact (mutations affect important sites, e.g. conserved sites)"
                       , "Combined", "Other (please describe)" = "Other")),
        conditionalPanel("input.IDstrategy == 'Other'", textInput("IDstrategy", "Describe")),
        radioButtons("cancer_type", "What types of cancer tested", 
                     c("Single cancer type", "Multiple cancer types", "Pan-Cancer analysis")),
        radioButtons("cofactors", "Does the method account for cofactors", 
                     c("Sequence trinucleotide content", "Gene expression level", 
                       "Protein disorder", "Gene replication", "Other (please describe)" = "Other",
                       "None")),
        conditionalPanel("input.cofactors == 'Other'", textInput("cofactors", "Describe")),
        radioButtons("limit", paste("Is the method limited to a subset of genes ",
                                      "(e.g ncRNAs, genes with phosphosites, genes ",
                                      "with 3D protein structures, etc)", sep = ""),
                     c("Yes (please describe)" = "Yes", "No")),
        conditionalPanel("input.limit == 'Yes'", textInput("limDesc", "Describe")),
        hr(),
        h3("Statistical Significance Testing"),
        radioButtons("method_pvals", "Does the method compute p-values",
                     c("Standartd test (T, Fisher's Exact, Wilcoxon, etc)",
                       "Model-based test (e.g. linear regression)",
                       "Sample-based test (permutation test, bootstrap", "No p-values computed")),
        radioButtons("supLearning", paste("Does method perform supervised learning using known ",
                                       "examples (support vector machine, naive bayes, ",
                                       "random forests, etc)", sep = ""),
                     c("Yes (please describe)" = "Yes", "No")),
        conditionalPanel("input.supLearning == 'Yes'", textInput("supLearningDesc", "Describe")),
        hr(),
        h3("Method Evaluation"),
        checkboxGroupInput('enrichment', paste("Enrichment of known cancer genes ",
                                               "among results (check one or more)", sep = ""),
                           c("Based on sequencing studies", "Based on complementary omics studies",
                             "Based on earlier literature", "None")),
        radioButtons("pnAnalysis", "Pathway & network analysis of resulting genes",c("Yes", "No")),
        radioButtons("litAnalysis", "Literature analysis of resulting genes", c("Yes", "No")),
        radioButtons("evalFDR", paste("Performance on randomised input data to ",
                                      "evaluate false discovery rate", sep = ""), c("Yes", "No")),
        radioButtons("evalRobustness", paste("Perfomance on random subsets of data ",
                                             "to test robustness", sep = ""), c("Yes", "No")),
        radioButtons("relPerformance", paste("Performance relative to other cancer ",
                                             "driver discovery methods", sep = ""), c("Yes", "No")),
        radioButtons("evalParam", "Evaluation of method parameters", c("Yes", "No")),
        radioButtons("evalDist", paste("Evaluation of resulting p-value distributions ",
                                       "(quantile-quantile analysis, etc)", sep = ""), c("Yes", "No")),
        actionButton("submit", "Submit", class = "btn-primary")
    )
)
